nemo/collections/nlp/models/machine_translation/transformer_mt_model.py

```diff
@@ -0,0 +1,332 @@
+        return ans
+
+    def validation_epoch_end(self, outputs):
+        """
+        Called at the end of validation to aggregate outputs.
+        :param outputs: list of individual outputs of each validation step.
+        """
+        self.log_dict(self.eval_epoch_end(outputs, 'val'))
+        # return self.eval_epoch_end(outputs, 'val')
+
+    def test_epoch_end(self, outputs):
+        return self.eval_epoch_end(outputs, 'test')
+
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
+
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config)
+        self.num_examples['val'] = val_data_config.get('num_examples', self.num_examples['val'])
+
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config)
+        self.num_examples['test'] = test_data_config.get('num_examples', self.num_examples['test'])
+
+    def _setup_dataloader_from_config(self, cfg: DictConfig):
+        dataset = TranslationDataset(
+            tokenizer_src=self.src_tokenizer,
+            tokenizer_tgt=self.tgt_tokenizer,
+            dataset_src=str(Path(cfg.src_file_name).expanduser()),
+            dataset_tgt=str(Path(cfg.tgt_file_name).expanduser()),
+            tokens_in_batch=cfg.tokens_in_batch,
+            clean=cfg.get("clean", False),
+            max_seq_length=cfg.get("max_seq_length", 512),
+            min_seq_length=cfg.get("min_seq_length", 1),
+            max_seq_length_diff=cfg.get("max_seq_length_diff", 512),
+            max_seq_length_ratio=cfg.get("max_seq_length_ratio", 512),
+        )
+        if cfg.shuffle:
+            sampler = pt_data.RandomSampler(dataset)
+        else:
+            sampler = pt_data.SequentialSampler(dataset)
+        return torch.utils.data.DataLoader(
+            dataset=dataset,
+            batch_size=1,
+            sampler=sampler,
+            num_workers=cfg.get("num_workers", 2),
+            pin_memory=cfg.get("pin_memory", False),
+            drop_last=cfg.get("drop_last", False),
+        )
+
+    def translate(self, text: List[str]) -> List[str]:
+        """
+        Translates list of sentences from source language to target language.
+        Should be regular text, this method performs its own tokenization/de-tokenization
+        Args:
+            text: list of strings to translate
+
+        Returns:
+            list of translated strings
+        """
+        res = []
+        for txt in text:
+            src = torch.tensor(self.src_tokenizer.text_to_ids(txt), device=self.device)
+            src = torch.unsqueeze(src, 0)
+            src_embeddings = self.src_embedding_layer(input_ids=src)
+            src_embeddings *= src_embeddings.new_tensor(self.emb_scale)
```

Comment:
Scaling the embeddings is commented out in the forward function L166 `# src_embeddings *= src_embeddings.new_tensor(self.emb_scale)`, but not here.

Reply:
The problem is our model is not training if I uncomment scaling.

Reply:
I see, but you probably want to comment it out here as well when doing inference right?

Reply:
I see. It is fixed.